packages/features/ee/workflows/pages/index.tsx

```diff
@@ -37,20 +45,66 @@ function WorkflowsPage() {
     },
   });
 
+  const query = trpc.viewer.workflows.getByViewer.useQuery();
+
+  useEffect(() => {
+    const allWorkflows = allWorkflowsData?.workflows;
+    if (allWorkflows && allWorkflows.length > 0) {
+      const filtered = allWorkflows.filter((workflow) => {
+        if (checkedFilterItems.teamIds.includes(workflow.teamId || 0)) return workflow;
+        if (!workflow.teamId) {
+          if (!!workflow.userId && workflow.userId === checkedFilterItems.userId) return workflow;
+        }
+      });
+      setFilteredWorkflows(filtered);
+    } else {
+      setFilteredWorkflows([]);
+    }
+  }, [checkedFilterItems, allWorkflowsData]);
+
+  useEffect(() => {
+    if (session.status !== "loading" && !query.isLoading) {
+      if (!query.data) return;
+      setCheckedFilterItems({
+        userId: session.data?.user.id || null,
+        teamIds: query.data.profiles
+          .map((profile) => {
+            if (!!profile.teamId) {
+              return profile.teamId;
+            }
+          })
+          .filter((teamId): teamId is number => !!teamId),
+      });
+    }
+  }, [session.status, query.isLoading, allWorkflowsData]);
+
+  if (!query.data) return null;
+
+  const profileOptions = query.data.profiles
+    .filter((profile) => !profile.readOnly)
+    .map((profile) => {
+      return { teamId: profile.teamId, label: profile.name || profile.slug, image: profile.image };
+    });
```

Comment:
We can map and filter in one iteration:

```suggestion
  const profileOptions = query.data.profiles.reduce<React.ComponentProps<typeof CreateButton>["options"]>(
    (profiles, profile) => {
      if (profile.readOnly) return profiles;
      profiles.push({ teamId: profile.teamId, label: profile.name || profile.slug, image: profile.image });
      return profiles;
    },
    []
  );
```